wpilibc/src/main/native/cpp/shuffleboard/ShuffleboardContainer.cpp

```diff
@@ -0,0 +1,167 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "frc/shuffleboard/ShuffleboardContainer.h"
+
+#include <wpi/raw_ostream.h>
+
+#include "frc/shuffleboard/ComplexWidget.h"
+#include "frc/shuffleboard/ShuffleboardComponent.h"
+#include "frc/shuffleboard/ShuffleboardLayout.h"
+#include "frc/shuffleboard/SimpleWidget.h"
+
+using namespace frc;
+
+ShuffleboardContainer::ShuffleboardContainer(wpi::StringRef title)
```

Comment:
Generally it's better to use `const wpi::Twine&` instead of `StringRef` for string parameters like this, to enable the user to do low-cost string concatenation at the callsite.  Twine provides a function to get a std::string for permanent storage, and the underlying NetworkTables interfaces have Twine inputs.

Reply:
Should we consider replacing all our StringRef stuff in wpilib with Twine at some point?

Reply:
Probably.  It's not *always* a better solution, but for input parameters it's usually a better choice.  It's no more expensive than StringRef and can easily be converted back to StringRef at the callsite (just need a local SmallString buffer in the case a copy is necessary).